FWCore/Services/scripts/edmTracerCompactLogViewer.py

```diff
@@ -203,24 +238,132 @@ def textPostfix(self):
+            results.append(container[i+1])
+            del container[i]
+            del container[i]
+            break
+    return results
+        
+            
 class PreFrameworkTransitionParser (FrameworkTransitionParser):
     def __init__(self, payload):
         super().__init__(payload)
     def textSpecial(self):
         return "starting"
+    def jsonInfo(self, counter, data):
+        if transitionIsGlobal(self.transition):
+            index = 0
+            if self.transition == Phase.startTracing:
+                data["globals"][0].append(jsonTransition(type=self.transition, id=index, sync=list(self.sync),start=0, finish=self.time ))
+                return
+            elif self.transition == Phase.esSync:
+                if self.sync[1] == 4294967295:
+                    #at end run transition
+                    index = findMatchingTransition(list(self.sync), data["globals"])
+                    container = data['globals'][index]
+                    container[-1]["finish"] = self.time*kMicroToSec
+                else:
+                    data['queued'][-1]["finish"] = self.time*kMicroToSec
+                    data['queued'].append( jsonTransition(type=self.transition, id = index, sync=list(self.sync), start=self.time , finish=0))
+                    return
+            elif self.transition==Phase.globalBeginRun:
+                index = self.index
+                #find associated es queued items
+                queued = data["queued"]
+                q = popQueuedTransitions(list(self.sync), queued)
+                globals = data['globals']
+                while index+1 > len(globals):
+                    globals.append([])
+                container = globals[index]
+                #find source, should be previous
+                last = container[-1]
+                if last["type"]==Phase.globalBeginRun and last["isSrc"]:
+                    last["sync"]=list(self.sync)
+                container.append(q[0])
+                container.append(q[1])
+            elif self.transition==Phase.globalBeginLumi:
+                index = self.index
+                #find associated es queued items
+                queued = data["queued"]
+                q = popQueuedTransitions(list(self.sync), queued)
+                globals = data['globals']
+                while index+1 > len(globals):
+                    globals.append([])
+                container = globals[index]
+                #find source, should be previous
+                last = container[-1]
+                if last["type"]==Phase.globalBeginLumi and last["isSrc"]:
+                    last["sync"]=list(self.sync)
+                container.append(q[0])
+                container.append(q[1])
+            elif Phase.globalEndRun == self.transition or self.transition==Phase.globalBeginLumi or Phase.globalEndLumi == self.transition or self.transition==Phase.globalWriteRun or Phase.globalWriteLumi == self.transition:
```

Comment:
What would you think about
```suggestion
            elif transition in [Phase.globalEndRun, Phase.globalBeginLumi, Phase.globalEndLumi, Phase.globalWriteRun, Phase.globalWriteLumi]:
```
?

Reply:
done